src/rust/pal/mod.rs

```diff
@@ -1,4 +1,17 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+//======================================================================================================================
+// PAL: Platform Abstraction Layer
+//======================================================================================================================
+
+// This is the platform abstraction layer designed to hide the platform specific implementation details of contants and
+// datastructures. This layer exists because of the differences between the Windows/Linux libc/WinSock implementations
+// of the corresponding Rust libraries.
+
+pub mod constants;
+pub mod data_structures;
+pub mod functions;
```

Comment:
I wonder if this is even the right way to do this.  Instead of having constants, data_structures, and functions, I would have a single module for each type of thing, like "networking" or "sockets".  Doing it this way won't scale well.

And the existing "linux" module should be converted to "shared_memory", honestly.

I have an outstanding PR where I'm adding an "arch" module to cover differences in CPU architectures.  It would make no sense for me to go to the "constants" module to add constants for that, and then "data_structures", etc. for example.

Reply:
@BrianZill I agree, but I asked @anandbonde to get first this merged in, and then he will re-work this source tree.

Reply:
Agreed, Pedro and I had planned to do this in subsequent PR(s). [As per the next steps listed in the PR description above: (1) Refactor the PAL to make platform/subsystem specific hierarchy in the directory structure.]

Reply:
I'll go ahead and agree to this, although it's not at all clear to me why you just can't change this now.  Once badly laid out structure makes it into the source tree, it can be hard to remove.  Using this very pal thing as an example, we currently have this "linux" module that shouldn't be there.  The differentiation by platform should only be in each individual #[cfg(whatever)] line.  I hope you're planning of fixing that too in your subsequent PR.